.ado/jobs/integration-test.yml

```diff
@@ -5,125 +5,160 @@ parameters:
+              displayName: Set BuildLogDirectory
+
+            - script: .ado\scripts\SetupLocalDumps.cmd integrationtest
+              displayName: Set LocalDumps
+
+            - ${{ if eq(matrix.UseHermes, 'true') }}:
+              - powershell: (Get-Content .\windows\ExperimentalFeatures.props) -replace 'false','true' | Set-Content .\windows\ExperimentalFeatures.props
+                displayName: Enable Hermes
+                workingDirectory: packages/integration-test-app
+
+            - ${{ if eq(matrix.BuildConfiguration, 'Debug') }}:
+              # The build is more likely to crash after we've started other bits that
+              # take up memory. Do the expensive parts of the build first.
+              - script: yarn windows --no-launch --no-packager --no-deploy --no-autolink --arch ${{ matrix.BuildPlatform }} --logging --buildLogDirectory $(BuildLogDirectory)
+                displayName: yarn windows --no-launch
+                workingDirectory: packages/integration-test-app
+
+              - powershell: Start-Process npm -ArgumentList "run","start"
+                displayName: Start packager
+                workingDirectory: packages/integration-test-app
+
+              # Instance load may time out if the packager takes too long to return a
+              # bundle. Request the bundle ahead of time so that the instance will get
+              # it back quickly.
+              - powershell: |
+                  while ($true) {
+                    try {
+                      Invoke-WebRequest -UseBasicParsing -Uri "http://localhost:8081/index.bundle?platform=windows&dev=true"
+                      break
+                    } catch [Exception] {
+                      echo $_.Exception
+                      Start-Sleep -s 1
+                    }
+                  }
+                displayName: Warm packager
+  
+              # If a debugger isn't connected to the packager, the CLI will launch the
+              # default browser to go there. On Windows Server that means launching IE
+              # with a bunch of restrictions. Explicitly launch Chrome to the debugger
+              # UI instead.
+              - powershell: Start-Process chrome http://localhost:8081/debugger-ui/
+                displayName: Launch debugger-ui
+
+              # There's a slight race condition here, where we assume Chrome will open
+              # before this launches the app. This step takes ~1m on CI machines so it
+              # shouldn't be a practical concern.
+              - script: yarn windows --no-build ${{ matrix.DeployOptions }} --no-packager --no-autolink --arch ${{ matrix.BuildPlatform }} --logging
+                displayName: yarn windows --no-build
+                workingDirectory: packages/integration-test-app
+
+            - ${{ if eq(matrix.BuildConfiguration, 'Release') }}:
```

Comment:
>- ${ [](http://example.com/codeflow?start=12&length=4)

Wont' thsi step do the same work of the preivous one? Should that one be in an `if ne(..., 'Release')`

Reply:
There are some differences in the series of steps and flags passed. In debug we run run-windows multiple times to first build, then later deploy + run.